### ISSUE TITLE
lda multicore not scaling into large number of cores

### ISSUE THREAD

**Issue by abhishekbuyt** [RaRe-Technologies/gensim#1592]
I believe it is due to the primary process being unable to keep up with both filling up the  queues and also merging the result. 

Will it not work better if we split the queue filling part of the code into the a separate process. 
If there are separate processes for queue filling and result queue processing. i dont see any issue. 

Note: i am sure it is not a IO issue as i dont see any IO load on using iostat and iotop

Or may be I should be usig some other corpus format which takes less time to deserialize. ??

Any guidance will help. thanks in advance

**Comment by menshikh-iv** [comment]
Hi @abhishekbuyt,
you can try to implement this, this will be a nice improvement for our LDA.

CC: @piskvorky

**Comment by piskvorky** [comment]
I agree, the queue filling can be intensive, converting documents into CSC matrices, IIRC. The merging should be fast though (again, IIRC, it's been a while since I saw the code).

A good first step seems to be profiling where the time actually goes. Verifying the bottlenecks. After that we can discuss the best ways to optimize that.
